### PR TITLE
Fix the Rust 1.51 panic warning

### PR DESCRIPTION
**Pull request by dabreegster**
The merge history in the PR looks kind of crazy, but I think if you commit as a squash, it'll turn out simple

### CODE REVIEW

File: src/fast_graph_builder.rs [hunk 1]
```diff
@@ -227,7 +227,7 @@ impl FastGraphBuilder {
                 return edge_id;
             }
         }
-        panic!["could not find out-edge id"]
+        panic!("could not find out-edge id")
```
Comment (dabreegster) [inline src/fast_graph_builder.rs:230]: I also noticed this. I guess technically the panic macro allows any kind of delimiter, but I've only ever seen `()` used

File: src/preparation_graph.rs [hunk 2]
```diff
@@ -145,13 +145,12 @@ impl PreparationGraph {
     }
 
     fn assert_valid_node_id(&self, node: NodeId) {
-        assert!(
-            node < self.num_nodes,
-            format!(
-                "invalid node id {}, must be in [0, {}[",
+        if node >= self.num_nodes {
+            panic!(
+                "invalid node id {}, must be in [0, {})",
```
Comment (dabreegster) [inline src/preparation_graph.rs:150]: At first I thought the second `[` was a typo, but I guess you meant the open interval. I learned `[]` as closed and `()` as open in school -- not sure if this is standard or not. I can revert back to `[` if you prefer.

File: src/fast_graph_builder.rs (hunk 1 above)
Comment (easbar) [reply src/fast_graph_builder.rs:230]: Oh wow, thanks for figuring this out, I could not make sense of the error message :+1:

File: src/preparation_graph.rs (hunk 2 above)
Comment (easbar) [reply src/preparation_graph.rs:150]: Ok, not sure about North America, but over here we learn either `[a,b[` or `[a,b)` as (half) open interval, it is equivalent. Anyway, either way is fine for me.

**Comment by easbar** [comment]
Thanks for the quick fix!

**Pull request merged**